split Cargo slides

listochkin opened the pull request:
Now we can pick and choose what part of the deck to present.

* Added a dedicated "Cargo" section.
* split the deck into 4: 
    * cargo commands and "features" 
    * dependency resolution, different kind of dependencies
    * cargo workspaces
    * rust project build time

Added some extra info:
* Speaker notes for Dependencies deck has a list of different private registry vendors and projects
* Workspaces deck has a shell function adapted from the one I use for trainings to set up a new workspace project

miguelraz commented on training-slides/src/rust-build-time.md:
```diff
@@ -0,0 +1,108 @@
+
+* Cargo keeps track of changes you make and only rebuilds what is necessary
+* When building a crate `rustc` can do most of work in parallel, but some steps still require synchronization
+* Depending on type of a build times spent in different build phases may be vastly different.
```
```suggestion
* Depending on the type of build, times spent in different build phases may be vastly different.
```

miguelraz commented:
Ok, I like having split up the Cargo deck to something a lot more manageable, nice work there.

I may be wrong here, but I think the `Cargo` heading in the Summary should be pushed to the bottom as a subchapter in the `Additional Material` part in case people ask for it - the way you've written it now, a trainee would see our Index on the left hand side and see
```
# Rust Fundamentals
# Applied Rust 
# Cargo 
# Additional Materials
```

I will bikeshed this and say it should instead be 

```
# Rust Fundamentals
# Applied Rust 
# Additional Materials
    ## Cargo
```

miguelraz commented:
Nit pick time:

We should decide if bullet points always start with an uppercase or lowercase letter. This PR alternates between both and we should pick a style.

My vote is for starting all bullet points with uppercase.

I'm guessing we're gonna need a style guide at a future point.

listochkin commented:
I don't really know about the bullet point style. I *like* seeing them the same everywhere, and yet once the slides are compiled and rendered by Reveal the uppercase bullet points seem too busy with capital letters almost jumping on top of each other. 

Meanwhile using lowercase everywhere feels very wrong sometimes, especially when the bullet point has multiple sentences:
* it starts with a lower case. And now the capital "A" looks very awkward.

jonathanpallant commented:
In English there are two kinds of bullet points:

* this kind, which continue the previous sentence,
* such that you could remove the bullets and it all still forms a valid sentence, or
* the other kind which I am going to show you next.

Versus:

* Here is the second kind
* Each is a sentence fragment
* They all start with a capital letter
* There is no continuation between them

jonathanpallant commented:
@listochkin are you able to get this one wrapped up?

listochkin commented:
I updated the summary page. `mdslides` doesn't support nested lists on summary page (and also doesn't support a few other features that I actually want, so I'll be sending some PRs there). Because of that for now I dumped all four decks into "Additional Material" section.

Also, changed all bullet points to lowercase initials for consistency. Looking back at the original version of slides I completely agree with Raz - capitalization was *wild*! I decided to keep capitalizing words like Rust, Cargo, C when they are at the beginning of a bullet. However, when they represent commands like `cargo` I keep them lowercase. In general lowercase works better for me visually: line lights in Reveal are very small for the font size and capitalized Bullet points feel too crammed.

Other than cosmetic updates the content is the same.

This should be ready for merging.

listochkin commented:
> @listochkin are you able to get this one wrapped up?

As a side note: GH notifications *do not work for me as a tool at all*. I only say your nudge after I finished with the code and went to write a note above. Ping me on Zulip if you want to get things bumped up priority-wise.

jonathanpallant commented:
What do you propose we put in the Roadmap for a standard training? Is there a "short version" we can give people so they know enough to be able to put simpleDB in a workspace? Or is the exercise enough to teach people that?

listochkin commented:
I think "Cargo Workspaces" is the only one I would move to Applied Rust

"Using Cargo" deck is mostly for completeness. A trainer is going to show most day-to-day commands to participants anyway, and everything else is in there only if there are further questions from a group (Like: "What are features?" or "Can Cargo do cross-platform builds?"). "Dependency Management with Cargo" and "Rust Projects Build Time" are similar.

"Workspaces" deck alone is tiny, and can be shown / shared before the exercise.
